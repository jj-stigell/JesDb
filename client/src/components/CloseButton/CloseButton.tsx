import React from 'react'

interface ICloseButton {
  onClick: () => void
  disabled?: boolean
}

export default function CloseButton ({
  onClick,
  disabled
}: ICloseButton): React.JSX.Element {
  return (
    <button
      onClick={onClick}
      disabled={disabled}
      type="button"
      aria-label="Close"
      className={
        `absolute top-3 end-2.5 text-gray-400 bg-transparent hover:bg-gray-200 hover:text-gray-900
        rounded-lg text-sm w-8 h-8 ms-auto inline-flex justify-center items-center
        dark:hover:bg-gray-600 dark:hover:text-white`
      }
      data-modal-hide="popup-modal"
    >
      <svg
        className="w-3 h-3"
        aria-hidden="true"
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 14 14"
      >
        <path
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
        />
      </svg>
    </button>
  )
}
