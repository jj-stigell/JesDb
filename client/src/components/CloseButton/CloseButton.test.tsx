import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import CloseButton from './CloseButton'

describe('CloseButton Component', () => {
  const mockOnClick = jest.fn()

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('renders correctly', () => {
    const { asFragment } = render(<CloseButton onClick={mockOnClick} />)

    expect(screen.getByRole('button')).toBeInTheDocument()
    expect(asFragment()).toMatchSnapshot()
  })

  it('calls onClick when the button is clicked', () => {
    render(<CloseButton onClick={mockOnClick} />)
    fireEvent.click(screen.getByRole('button'))

    expect(mockOnClick).toHaveBeenCalledTimes(1)
  })

  it('does not call onClick when the button is disabled', () => {
    const { asFragment } = render(<CloseButton onClick={mockOnClick} disabled />)
    fireEvent.click(screen.getByRole('button'))

    expect(mockOnClick).not.toHaveBeenCalled()
    expect(asFragment()).toMatchSnapshot()
  })
})
