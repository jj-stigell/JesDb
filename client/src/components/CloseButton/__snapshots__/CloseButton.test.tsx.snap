// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton Component does not call onClick when the button is disabled 1`] = `
<DocumentFragment>
  <button
    aria-label="Close"
    class="absolute top-3 end-2.5 text-gray-400 bg-transparent hover:bg-gray-200 hover:text-gray-900
        rounded-lg text-sm w-8 h-8 ms-auto inline-flex justify-center items-center
        dark:hover:bg-gray-600 dark:hover:text-white"
    data-modal-hide="popup-modal"
    disabled=""
    type="button"
  >
    <svg
      aria-hidden="true"
      class="w-3 h-3"
      fill="none"
      viewBox="0 0 14 14"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`CloseButton Component renders correctly 1`] = `
<DocumentFragment>
  <button
    aria-label="Close"
    class="absolute top-3 end-2.5 text-gray-400 bg-transparent hover:bg-gray-200 hover:text-gray-900
        rounded-lg text-sm w-8 h-8 ms-auto inline-flex justify-center items-center
        dark:hover:bg-gray-600 dark:hover:text-white"
    data-modal-hide="popup-modal"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="w-3 h-3"
      fill="none"
      viewBox="0 0 14 14"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
      />
    </svg>
  </button>
</DocumentFragment>
`;
